### PR TITLE
Tests: add new RootPackageHandlingTest + bugfix

### LINKED ISSUE #32: Invalid "installed_paths" config value
## Problem/Motivation

Relates to v0.4.0

PHPCodeSniffer's `installed_paths` config value is set incorrectly.

Two problems:
- `frenck/php-compatibility` is duplicated: once it's absolute, once it's relative path
- Paths contain directories different than `phpcodesniffer-standard` type of package (`../../doctrine`)

## Expected behaviour

Two problems:
- Paths should contain unique values
- Paths should contain only packages with `phpcodesniffer-standard` type

## Actual behaviour

```
-> % composer run-script install-codesniffs
> Dealerdirect\Composer\Plugin\Installers\PHPCodeSniffer\Plugin::run
PHP CodeSniffer Config installed_paths set to <path_to_project>/vendor/frenck/php-compatibility,../../doctrine/,../../frenck/php-compatibility/
```

## Steps to reproduce

Require `doctrine/doctrine-cache-bundle` in Composer.

## Proposed changes

Get rid of `$searchPaths = [getcwd()];`, should fix both issues.

### PR DESCRIPTION
**Pull request by jrfnl**
### TestCase: new recursiveDirectoryCopy() method

New helper method to recursively copy a (fixture) directory with all its files to another location, typically, one of the system temp test directories.

If the `$srcDir` contains a `composer.json` file, it will get the same `TestCase::writeComposerJsonFile()` special treatment, meaning that the artifacts directory will be injected, as well as plugin permissions and more.

### Tests: add new RootPackageHandlingTest

This new test class tests that the plugin correctly registers standards found in the root package if it is an external standard.

Notes:
* This test is about Composer and the plugin, so does not need to be tested against multiple PHPCS versions.

This test class covers the following bugs previously reported:
* Dealerdirect/phpcodesniffer-composer-installer#19
* Dealerdirect/phpcodesniffer-composer-installer#21
* Dealerdirect/phpcodesniffer-composer-installer#20
* Dealerdirect/phpcodesniffer-composer-installer#25
* Dealerdirect/phpcodesniffer-composer-installer#32

### Plugin: only search root package when it has the `phpcodesniffer-standard` type (bug fix)

In the "olden days" of PHPCS 1.x, it was customary for projects to call their custom project ruleset `ruleset.xml` instead of `[.]phpcs.xml[.dist]`.

This could lead to a root package with such an _old-style_ PHPCS ruleset being registered as if it were a proper PHPCS external standard.
This was previously reported in issue #32.

The fix I'm proposing now, applies the same rules as for `vendor` installed standards to the root package, i.e.:
* Must have the `phpcodesniffer-standard` type set in the `composer.json` file AND
* Must have a `ruleset.xml` file.

Root packages which do not comply with _both_ these rules will no longer be considered for registration with PHPCS. This should also make the plugin slightly faster for those packages which do not have external standards, but do have the plugin installed.

Includes removing some dead code (condition which could never be `true`) which was loosely related to this.

Fixes #32

## Related Issues

Related to #92

### CODE REVIEW

**Review by Potherca — DISMISSED** [review]

File: tests/TestCase.php
Comment (jrfnl) [reply tests/TestCase.php:573]: Good question and I honestly don't remember why I went with `file_exists()` anymore. There most likely was a reason, but as I can't remember it, let's change this to `is_dir()`. If needs be, we can always change it back again at a later point in time.

I'm also seeing a `realpath()` being used without checking that it doesn't result in `false` in this function (though it shouldn't anyway what with the `file_exists()`/`is_dir()` check before it).
As this is code only used in the tests, I'm going to leave that for now.

Comment (jrfnl) [reply tests/TestCase.php]: Rebased & force pushed. Switching out `file_exists()` for `is_dir()` is the only change.

**Review by Potherca — APPROVED** [review]

**Comment by jrfnl** [comment]
Rebased without changes after the merge of #180 to get a clean and passing build. Will merge after the build passes.

**Pull request merged**
